### PR TITLE
chore: added translations for es_ES

### PR DESCRIPTION
**Pull request by gboquizosanchez**
To continue with #2767. I have added Spanish (Spain) localization. I have followed docs contained in #2765.

### CODE REVIEW

**Comment by wescopeland** [comment]
Hi @gboquizosanchez! Thanks for contributing. Unless anyone voices any opinions about the strings in the file over the next few days, I plan on merging this.

Important to note: es_ES will not initially be available as a selectable locale. Before it can be made available, it also needs to be hooked up into the number and date l10n utils so those values display properly on the initial React.js server render. It's still early days with our i18n/l10n stuff :)

I have a few tasks I'm in the middle of finishing up. Afterwards, I'll be happy to hook es_ES up into our formatter utils. I'll also add a reference to that PR into the translations doc so the process is smoother for new locales going forward.

File: lang/es_ES.json
Comment (wescopeland) [inline lang/es_ES.json:30]: I am wondering if "logros" is a typo. "achievers" in this context represents the number of players who have unlocked the achievement, not the count of achievements.

Comment (gboquizosanchez) [reply lang/es_ES.json:30]: Probably is a typo, I'll check it asap. I think "desbloqueadores" or similar.

Comment (gboquizosanchez) [reply lang/es_ES.json:30]: Humm, the translation of this is "complicated" beacuse "desbloqueadores de logros" is redundant. Achiever is a person that has success, and in this case has unlocked an achievement. But we don't have an appropiate translation for this term...

If I put this into the translator DeepL. It suggests me those translations or alternatives.

image

I suggest "número de usuarios que han desbloqueado logros" but it's excessive long. I'm really confused about the correct term for this.

Comment (gboquizosanchez) [reply lang/es_ES.json]: I have pushed a new translation for this. @wescopeland.

**Review by wescopeland — APPROVED** [review]

**Comment by wescopeland** [comment]
Hi @gboquizosanchez! Happy to take what you already have. Not sure if you'd like to go over the latest en_US file before or after merging this one. With the introduction of the <Trans /> component, several keys have changed.

**Comment by gboquizosanchez** [comment]
I think I prefer to change those new keys before you merge this PR. So, I'll start asap. Thanks for the notice!

File: lang/es_ES.json
Comment (WedgeSparda) [inline lang/es_ES.json:73]: Hi, I was working on another spanish translation before discovering about you work. So I'll drop my branch since you got here first.

Just a point, what do you think about using `¿Realmente quieres` instead of `¿Estás seguro de que quieres` in all confirmation prompts. That's how I was setting those translations in my branch. The reason behind that choose of words was to prompt a message without distinction of genre and also a bit shorter.

What do you think?

Comment (gboquizosanchez) [reply lang/es_ES.json:73]: Hi @WedgeSparda, I've added the remaining keys after @wescopeland merged the new <Trans> component that includes advanced translation support.

I've incorporated some of the changes you introduced in your PR #2791 because they improved upon my original translation.

Regarding the phrase _Realmente quieres_, I acknowledge it's more concise, but _Estás seguro de que_ might be a better fit. If it appears odd on the website, we can consider altering it. However, I believe this version is more effectively localized.

What are your thoughts? Should we conduct a direct test to see which option performs better on the website?

File: lang/es_ES.json
Comment (WedgeSparda) [inline lang/es_ES.json:189]: English text remains

File: lang/es_ES.json
Comment (WedgeSparda) [reply lang/es_ES.json:73]: If you see is a better fit, we can go with `Estás seguro de que` my only concern was the genre, but I think we're all used to that message.

File: lang/es_ES.json
Comment (WedgeSparda) [inline lang/es_ES.json:296]: Jump line

File: lang/es_ES.json
Comment (gboquizosanchez) [reply lang/es_ES.json:189]: image

I don't got it.

File: lang/es_ES.json
Comment (wescopeland) [reply lang/es_ES.json:296]: FWIW I see this as well. The line count in this file should match en_US in `master`.

Screenshot 2024-10-29 at 12 03 03 PM

Comment (gboquizosanchez) [reply lang/es_ES.json:296]: I'm checking my last commits because I see it's not sync. :\

Comment (gboquizosanchez) [reply lang/es_ES.json:296]: image

It's seems a GitHub issue or something... Now it's showing the good one.

File: lang/es_ES.json
Comment (gboquizosanchez) [reply lang/es_ES.json:73]: We can make RA more inclusive; however, this requires consideration in all the texts we translate.

What do you think? Should we change?

Comment (WedgeSparda) [reply lang/es_ES.json:73]: I would say yes. We should review future translations to match the current ones, not just in this specific topic.

Comment (gboquizosanchez) [reply lang/es_ES.json:73]: Alright, I will make the changes then.

Comment (WedgeSparda) [reply lang/es_ES.json]: Thank you

File: lang/es_ES.json
Comment (wescopeland) [inline lang/es_ES.json:99]:
```suggestion
    "Must be at least 8 characters.": "Debe tener al menos 8 caracteres",
```

File: lang/es_ES.json
Comment (wescopeland) [inline lang/es_ES.json:194]:
```suggestion
    "We will only retain personal information as long as necessary for the fulfillment of those purposes.": "Solo mantendremos información personal mientras sea necesario para el cumplimiento de dichos propósitos.",
```

File: lang/es_ES.json
Comment (wescopeland) [inline lang/es_ES.json:227]: Possible typo here?

**Review by wescopeland — COMMENTED** [review]
Just doing a final review! Spotted a few things worth pointing out.

File: lang/es_ES.json
Comment (WedgeSparda) [inline lang/es_ES.json:227]: I think it should be:

```suggestion
    "Comments on an achievement I created": "Comentarios en un logro creado por mí",
```

**Comment by wescopeland** [comment]
I will open a follow-up PR to make es_ES selectable and compatible with the localization utils for numbers and dates.

**Pull request merged**
